AstSemantics.md

```diff
@@ -154,6 +151,28 @@ by [available resources](Nondeterminism.md)). In the future, to support
 [>4GiB linear memory](FutureFeatures.md#heaps-bigger-than-4gib), support for
 indices with type `int64` will be added.
 
+### Adjusting memory size and permissions: `mmap`
+
+In the MVP the size of linear memory is fixed: The initial size of linear
+memory will remain unchanged for the life of that WebAssembly module. Later,
+we will support a limited form of `mmap` which can:
+
+  * Allocate pages of memory, in order to increase or decrease
+    the amount of available memory to the WebAssembly module.
+  * Adjust the permissions of a page of memory, for example to make
+    small effective addresses close to `0` behave as if they are
+    [out-of-bounds](AstSemantics.md#out-of-bounds)
+    (see [discussion](https://github.com/WebAssembly/design/issues/204)).
+
+In addition to the `mmap` operation, applications will also have access to
+
+ * `munmap`, to unmap `mmap`ed pages, and
+ * `mmap_get_page_size`, to detect the proper system page size.
```

Comment:
I don’t think this answers my question.

I think we should be clear about whether future performant wasm implementations are required to use page-level protections and large reservations.  Do you believe that this is the case?

-Filip


> On Jul 28, 2015, at 3:17 PM, JF Bastien <notifications@github.com> wrote:
> 
> In AstSemantics.md <https://github.com/WebAssembly/design/pull/285#discussion_r35708365>:
> 
> > +
> > +In the MVP the size of linear memory is fixed: The initial size of linear
> > +memory will remain unchanged for the life of that WebAssembly module. Later,
> > +we will support a limited form of `mmap` which can:
> > +
> > +  * Allocate pages of memory, in order to increase or decrease
> > +    the amount of available memory to the WebAssembly module.
> > +  * Adjust the permissions of a page of memory, for example to make
> > +    small effective addresses close to `0` behave as if they are
> > +    [out-of-bounds](AstSemantics.md#out-of-bounds)
> > +    (see [discussion](https://github.com/WebAssembly/design/issues/204)).
> > +
> > +In addition to the `mmap` operation, applications will also have access to
> > +
> > + * `munmap`, to unmap `mmap`ed pages, and
> > + * `mmap_get_page_size`, to detect the proper system page size.
> I'd keep the name vague for now, since it's related to feature detection.
> Also mention it's a "good enough" guess, not necessarily the actual value.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/WebAssembly/design/pull/285/files#r35708365>.
>